tests/dialects/test_tosa.py

```diff
@@ -0,0 +1,29 @@
+import pytest
+
+from xdsl.dialects.builtin import (
+    TensorType,
+    f32,
+    i32,
+)
+from xdsl.dialects.tosa import AddOp
+from xdsl.utils.exceptions import VerifyException
+from xdsl.utils.test_value import create_ssa_value
+
+tensor_i = create_ssa_value(TensorType(i32, [1, 2, 3, 4]))
+tensor_flat = create_ssa_value(TensorType(i32, [1, 1, 1, 1]))
+tensor_f = create_ssa_value(TensorType(f32, [1, 2, 3, 4]))
+
+
+def test_valid_cases():
+    AddOp(operands=[tensor_i, tensor_i], result_types=[tensor_i.type]).verify_()
```

Comment:
The idea is that other ops should reuse the same verification helper, so I would recommend creating a new function that takes three tensor types and an operation name and raises the appropriate exception.

Reply:
Ah sorry you added one! Can you please move the invalid cases into a filecheck test called `ops_invalid.mlir` with `--verify-diangostics`, and remove this pytest?